src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -195,14 +196,24 @@ private function createRoleHierarchy(array $config, ContainerBuilder $container)
     private function createAuthorization(array $config, ContainerBuilder $container)
     {
         foreach ($config['access_control'] as $access) {
-            $matcher = $this->createRequestMatcher(
-                $container,
-                $access['path'],
-                $access['host'],
-                $access['port'],
-                $access['methods'],
-                $access['ips']
-            );
+            if ($access['request_matcher']) {
+                if (!$container->has($access['request_matcher'])) {
+                    throw new InvalidConfigurationException('The request_matcher reference you passed to one of the access control items does not exist. Have you made a typo?');
+                }
+                if (!$container->get($access['request_matcher']) instanceof RequestMatcherInterface) {
+                    throw new InvalidConfigurationException('The request_matcher reference you passed to one of the access control items references a service that does not implement the RequestMatcherInterface. Did you forget to add it?');
+                }
```

Comment:
We can't do these checks here.

The `$container` property in DI extensions only contains a very small subset of the real container (and we're in the process of building the container in a DI extension, so the real container is incomplete).

Looking at the `request_matcher` option for firewalls, it seems like we don't do any validation there. Let's do the same here.

Reply:
Fixed :-)